### PR TITLE
[bitnami/fluentd] Add Fluentd Helm Chart tests

### PR DESCRIPTION
**Pull request by superaleks**
**Description of the change**
Increasing the test coverage of the Fluentd Helm Chart by adding Goss tests. 

**Benefits**

Ensuring higher quality of the Helm charts.

**Possible drawbacks**
Automated tests could show to be potentially flaky. 

**Test results**

:gear: - [PASS](https://github.com/superaleks/charts/pull/39)

### Checklist

- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in the agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: .vib/fluentd/goss/goss.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+
+file:
+  {{ printf "/opt/bitnami/fluentd/conf/%s" .Env.FLUENTD_CONF }}:
```
Comment (FraPazGal) [inline .vib/fluentd/goss/goss.yaml:3]: I think we could replace this test with a dry-run check of the config file using the `fluentd` binary as we do in BTS. The command is as follows:

```
$ fluentd -c /opt/bitnami/fluentd/conf/fluentd.conf --dry-run
2022-07-08 14:38:33 +0000 [info]: parsing config file is succeeded path="/opt/bitnami/fluentd/conf/fluentd.conf"
2022-07-08 14:38:34 +0000 [info]: gem 'fluentd' version '1.15.0'
...
```

File: bitnami/fluentd/templates/forwarder-clusterrole.yaml [hunk 2]
```diff
@@ -28,7 +28,7 @@ rules:
     resources:
       - "podsecuritypolicies"
     resourceNames:
-      - {{ include "common.names.fullname" . }}-forwarder
+      - {{ printf "%s-%s" (include "common.names.fullname.namespace" .) "-forwarder" }}
```
Comment (joancafom) [inline bitnami/fluentd/templates/forwarder-clusterrole.yaml:31]: Small typo, you would be adding an extra dash:

```suggestion
      - {{ printf "%s-%s" (include "common.names.fullname.namespace" .) "forwarder" }}
```

File: bitnami/fluentd/templates/forwarder-psp.yaml [hunk 3]
```diff
@@ -3,7 +3,7 @@
 apiVersion: policy/v1beta1
 kind: PodSecurityPolicy
 metadata:
-  name: {{ include "common.names.fullname" . }}-forwarder
+  name: {{ printf "%s-%s" (include "common.names.fullname.namespace" .) "-forwarder" }}
```
Comment (joancafom) [inline bitnami/fluentd/templates/forwarder-psp.yaml:6]: Same here

**Review by joancafom — CHANGES_REQUESTED** [review]
Thanks for the changes @superaleks ! I have added minor comments, please take a look when you have a chance 😁

File: bitnami/fluentd/templates/forwarder-clusterrole.yaml (hunk 2 above)
Comment (superaleks) [reply bitnami/fluentd/templates/forwarder-clusterrole.yaml]: Good catch! I've fixed it now.

File: .vib/fluentd/goss/goss.yaml (hunk 1 above)
Comment (superaleks) [reply .vib/fluentd/goss/goss.yaml]: Applied this improvement!

File: .vib/fluentd/goss/goss.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    filetype: file
+    exists: true  
+command:
+  fluentd-conf-test:
+    exec: fluentd -c /opt/bitnami/fluentd/conf/fluentd.conf --dry-run
+    exit-status: 0
+    stdout: 
+      - {{ printf "/bitnami.*%s/" .Env.BITNAMI_APP_NAME }}
+      - {{ printf "/port *%s/" .Env.FLUENTD_AGGREGATOR_SERVICE_PORT }}
+      - {{ printf "/port *%s/" .Env.FLUENTD_AGGREGATOR_SERVICE_PORT_TCP }}
```
Comment (joancafom) [inline .vib/fluentd/goss/goss.yaml:14]: After briefly checking the chart's design, I don't it makes sense to templatize this (or maybe even check)

The `fluentd.conf` file is not generated dynamically, but rather set to a fixed value:

https://github.com/bitnami/charts/blob/819db25f92b96b0a883cdae95228a961e4d83e35/bitnami/fluentd/values.yaml#L166-L170

Hence, even if the chart is configured with another service_port, the file won't be updated automatically.

Comment (superaleks) [reply .vib/fluentd/goss/goss.yaml]: OK, then it makes more sense to have this test check only the presence of the config file.

File: .vib/fluentd/goss/goss.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  /opt/bitnami/fluentd/Gemfile:
+    mode: "0644"
+    filetype: file
+    exists: true  
+command:
+  fluentd-conf-test:
+    exec: fluentd -c /opt/bitnami/fluentd/conf/fluentd.conf --dry-run
+    exit-status: 0
+    stdout: 
+      - {{ printf "/bitnami.*%s/" .Env.BITNAMI_APP_NAME }}
```
Comment (FraPazGal) [inline .vib/fluentd/goss/goss.yaml:12]: Do we know what the command outputs when it fails? The checked stdout may be too general. I would personally also look for the `parsing ... succeded` words or something similar.

**Review by FraPazGal — COMMENTED** [review]
Added a couple of comments, could you please check them?

File: .vib/fluentd/goss/goss.yaml (hunk 5 above)
Comment (superaleks) [reply .vib/fluentd/goss/goss.yaml:12]: Okay, added some more checks here.

**Review by FraPazGal — APPROVED** [review]

**Review by joancafom — APPROVED** [review]
LGTM!

**Review by agomezmoron — APPROVED** [review]
LGTM

**Comment by joancafom** [comment]
Closed in favor of https://github.com/bitnami/charts/pull/11427

**Pull request closed without merging**